scrypto/src/component/stubs.rs

```diff
@@ -234,25 +239,26 @@ extern_blueprint_internal! {
         fn instantiate(
             owner_role: OwnerRole,
             pool_manager_rule: AccessRule,
-            resource_addresses: Vec<ResourceAddress>,
+            resource_addresses: Vec<FungibleResourceManager>,
             address_reservation: Option<GlobalAddressReservation>,
         ) -> Global<MultiResourcePool>;
     },
     {
-        fn contribute(&mut self, buckets: Vec<Bucket>) -> (Bucket, Vec<Bucket>);
-        fn redeem(&mut self, bucket: Bucket) -> Vec<Bucket>;
-        fn protected_deposit(&mut self, bucket: Bucket);
+        fn contribute(&mut self, buckets: Vec<FungibleBucket>)
```

Comment:
I think there was some product requirement to support treating non-fungible in a fungible way. @0xOmarA what do you think?

Reply:
AFAIK pools of non-fungibles was not a product requirement and is actually something that the pools currently reject: https://github.com/radixdlt/radixdlt-scrypto/blob/7dbe606bac3d2661a47c30a43dafb5016074f130/radix-engine/src/blueprints/pool/v1/v1_0/one_resource_pool_blueprint.rs#L32-L34